lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/system/MongoOrganizationDao.java

```diff
@@ -129,7 +129,7 @@ public void deleteOrganization(final String organizationCode) {
    }
 
    @Override
-   public Organization editOrganization(final String organizationCode, final Organization organization) {
+   public Organization updateOrganization(final String organizationCode, final Organization organization) {
       MongoOrganization mongoOrganization = new MongoOrganization(organization);
       String id = datastore.createQuery(MongoOrganization.class)
                .field(MongoOrganization.CODE).equal(organizationCode).get().getId();
```

Comment:
You do not have to get the organization again from the database. It should have been already retrieved by `OrganizationFacade` so this database call is redundant.